File: ios/Mobile Client/Mobile Client/entities/AuditType.swift
//
//  AuditType.swift
//  Mobile Client
//
//  Created by Eric Ruck on 2/20/18.
//  Copyright © 2018 AuditPro. All rights reserved.
//

import Foundation


/**
 * Provides the valid audit types.
 * @author Eric Ruck
 */
class AuditType {
	static let STANDARD = AuditType(id: 1, name: "Standard")
	static let DEMO     = AuditType(id: 2, name: "Demo")
	static let RESEARCH = AuditType(id: 3, name: "Research")
	static let Types = [STANDARD, DEMO, RESEARCH]

	/**
	 * Initializes a new audit type
	 * @param id Unique audit type identifier
	 * @param name Audit type display name
	 */
	private init(id: Int, name: String) {
		self.id = id;
		self.name = name;
	}

	/** Provides the unique identifier of this audit type. */
	let id: Int

	/** Provides the display name of this audit type. */
	let name: String
}

File: android/mobileclient/app/src/main/java/com/auditpro/mobile_client/entities/AuditType.java
/*
  AuditPRO Mobile Client Android
  Copyright 2018 AuditPRO All Rights Reserved
 */
package com.auditpro.mobile_client.entities;


/**
 * Provides the valid audit types.
 * @author Eric Ruck
 */
public class AuditType {
	public static final AuditType STANDARD = new AuditType(1, "Standard");
	public static final AuditType DEMO = new AuditType(2, "Demo");
	public static final AuditType RESEARCH = new AuditType(3, "Research");
	public static final AuditType[] Types = { STANDARD, DEMO, RESEARCH };

	/**
	 * Initializes a new audit type
	 * @param id Unique audit type identifier
	 * @param name Audit type display name
	 */
	private AuditType(int id, String name) {
		this.id = id;
		this.name = name;
	}

	/**
	 * Gets the unique identifier of this audit type.
	 * @return Audit type unique identifier
	 */
	public int getId() {
		return id;
	}

	/**
	 * Gets the display name of this audit type.
	 * @return Audit type display name
	 */
	public String getName() {
		return name;
	}

	/** Provides the unique identifier of this audit type. */
	private int id;

	/** Provides the display name of this audit type. */
	private String name;
}
